File: Jewix-API/src/main/java/com/aomsir/jewixapi/pojo/entity/Article.java
package com.aomsir.jewixapi.pojo.entity;

import lombok.Data;

/**
 * @Author: Aomsir
 * @Date: 2023/2/24
 * @Description: 文章实体类
 * @Email: info@say521.cn
 * @GitHub: <a href="https://github.com/aomsir">GitHub</a>
 */
@Data
public class Article extends BaseEntity {
    /**
     * 主键id
     */
    private Long id;

    /**
     * 文章uuid
     */
    private String uuid;

    /**
     * 文章标题
     */
    private String title;

    /**
     * 文章封面
     */
    private String cover;

    /**
     * 文章内容
     */
    private String content;

    /**
     * 类型
     */
    private Integer type;

    /**
     * 文章源地址
     */
    private String originUrl;

    /**
     * 文章浏览量
     */
    private Integer views;

    /**
     * 是否置顶
     */
    private Integer isTop;

    /**
     * 是否逻辑删除
     */
    private Integer isDelete;

    /**
     * 文章描述
     */
    private String description;

}

File: Jewix-Front/src/servers/models/article.ts
import { HomeProps } from "@/pages"
import { BaseEntity, createBaseModel } from "."
// 后端文章模型
export interface Article extends BaseEntity {
    id: number
    uuid: string
    title: string
    cover: string
    content: string
    type: number
    originUrl: string
    views: number
    isTop: number
    isDelete: number
    description?: string
}
// HACK 后期可能会去掉非article的属性
/**
 * 创建文章模型，携带的参数则会覆盖默认值
 */
export const createArticle = <T extends Partial<Article>>(article?: T) =>
    ({
        id: 0,
        uuid: "",
        title: "",
        cover: "",
        content: "",
        type: 0,
        originUrl: "",
        views: 0,
        isTop: 0,
        isDelete: 0,
        ...createBaseModel(),
        ...article,
    } as T & Article)
// 模型转换
createArticle.toHomeArticle = (article: Partial<Article>): HomeProps["initialBlogs"][0] => {
    const _article = createArticle(article)

    return {
        id: _article.uuid,
        image: _article.cover,
        title: _article.title,
        desc: _article.description ?? _article.content,
        viewers: _article.views,
        isTop: !!_article.isTop,
    }
}
